pennylane/devices/qubit_mixed/sampling.py

```diff
@@ -133,12 +125,12 @@ def _measure_with_samples_diagonalizing_gates(
     total_indices = _get_num_wires(state, is_state_batched)
     wires = qml.wires.Wires(range(total_indices))
 
-    def _process_single_shot(samples):
+    def _process_single_shot_copy(samples):
```

Comment:
Why did you rename this function from `process_single_shot` to `process_single_shot_copy`?

Reply:
https://github.com/PennyLaneAI/pennylane/pull/6639#discussion_r1868040119